### PR TITLE
Makes Ephedrine not give spasms when holding a small or tiny item in your active hand, Also makes the movement speed scale with purity

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request

Makes Ephedrine not drop small and tiny items, Also makes the movement speed scale with purity

## Why It's Good For The Game

Ephedrine right now is really hard to use as an escape tool cause you drop things usually escaped for running away like tools, teleporters etc, This makes it better as a tool for that without making it too powerful for combat, Also makes movement speed scale with purity to make it more meaningful

## Changelog
:cl:
balance: Ephedrine spasms won't affect small or tiny items
balance: Ephedrine's movement speed bonus now scales with purity.
/:cl:

### CODE REVIEW

**Comment by Pickle-Coding** [comment]
Pretty sure the purpose of the balance change was to make ephedrine an escape plan rather than a combat chemical.

**Comment by starrm4nn** [comment]
> Pretty sure the purpose of the balance change was to make ephedrine an escape plan rather than a combat chemical.

Yeah and no one used it since then, The spasms happened way too often

**Comment by Higgin** [comment]
20u ephed is a lot, and you actually need more than that for an effect to offset natural organ healing. That's like 200s of full speed ahead baton immunity.

Organ damage unfortunately sucks and is pretty much binary. That's one of its weaknesses as a balance lever. 

You can't really drop the OD without risking making it a one-tap heart attack weapon. 

The higher speed and jitters make it better as an escape than an ALL COMBAT ALL THE TIME chem rn. 

Not sure we should drop that.

**Comment by starrm4nn** [comment]
> 20u ephed is a lot, and you actually need more than that for an effect to offset natural organ healing. That's like 200s of full speed ahead baton immunity.
> 
> Organ damage unfortunately sucks and is pretty much binary. That's one of its weaknesses as a balance lever.
> 
> You can't really drop the OD without risking making it a one-tap heart attack weapon.
> 
> The higher speed and jitters make it better as an escape than an ALL COMBAT ALL THE TIME chem rn.
> 
> Not sure we should drop that.

also natural organ healing takes quite a while, even with food that contains vitamins and rest, I still think the organ damage limits its usage as a resource compared to how it was used constantly before the nerf
also using it as an escape tool in practice was really difficult because you'd drop even non combat related items when trying to run away

**Comment by ATH1909** [comment]
> also using it as an escape tool in practice was really difficult because you'd drop even non combat related items when trying to run away

what if the spasms were returned to their old rate, but they were changed to only drop normal or larger items? so things like a hand tele or an id card wouldn't be dropped, but a baton or gun would.

**Comment by moocowswag** [comment]
Making it heart and lung damage = slime people buff

**Comment by Higgin** [comment]
> Not sure we should drop that.
> 
> natural organ healing takes quite a while, even with food that contains vitamins and rest, I still think the organ damage limits its usage as a resource compared to how it was used constantly before the nerf also using it as an escape tool in practice was really difficult because you'd drop even non combat related items when trying to run away

i hate to say it but organ damage sucks. it's pretty much binary: either your shit is nonfunctional or it's not. that initial 200s of speed and baton KD immunity are huge, and you can let your shit heal while looting your victims or waiting in hiding before you do it again. 

it's sort of the same problem a lot of our cat2 meds have - organ damage (except with sensory organs) really causes no meaningful problems before then. it's only useful as a balance lever against having the meds in you for a really long time, and even then, several of them only deal organ damage when they do their actual healing effect.

i'm sort of ambivalent to this change overall. it's fine, will probably be fine - hell, even if you brought it back at old ephed speeds. I think this is probably better than doing that, though it'd be better still to do the thing mentioned about making it only force drop items of a larger size class than small - or, make it only check to drop when you click on something with the item.

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 1]
```diff
@@ -596,14 +596,17 @@
...
-	if(SPT_PROB(10 * (1.5-creation_purity), seconds_per_tick) && iscarbon(affected_mob))
+	if(SPT_PROB(10 * (1-creation_purity), seconds_per_tick) && iscarbon(affected_mob))
 		var/obj/item/I = affected_mob.get_active_held_item()
 		if(I && affected_mob.dropItemToGround(I))
 			to_chat(affected_mob, span_notice("Your hands spaz out and you drop what you were holding!"))
 			affected_mob.set_jitter_if_lower(20 SECONDS)
 
 	affected_mob.AdjustAllImmobility(-20 * REM * seconds_per_tick * normalise_creation_purity())
 	affected_mob.adjustStaminaLoss(-1 * REM * seconds_per_tick * normalise_creation_purity(), updating_stamina = FALSE)
+	affected_mob.adjustOrganLoss(ORGAN_SLOT_HEART, 1 * REM * seconds_per_tick)
```
Comment (SyncIt21) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm:607]: Make it scale with normalise_creation_purity(). 

Also heart and lung is excessive. Make it apply only for lung as you could end up killing the guy again with just 30u of this stuff

**Review by SyncIt21 — CHANGES_REQUESTED** [review]

**Comment by ExcessiveUseOfCobblestone** [comment]
The chem MUST have an immediate downside to it that makes it unwieldy for combat, longer term downsides (such as organ damage) revert the intent of the chem which was to be used as an escape, not a speed boost during combat. In any case, the nuka cola speed is just not happening, it isnt explained anywhere as to why it needs that.

The design around the chem should be you dont want to take anymore than you have to in order to get out of dodge.

Also the chem seeing use or not is not a metric of the chem's success. By nature of the chem, you dont really expect it to see much use since it implies youre looking at a losing fight.

**Comment by moocowswag** [comment]
> Also the chem seeing use or not is not a metric of the chem's success. By nature of the chem, you dont really expect it to see much use since it implies youre looking at a losing fight.

A losing fight that you can predict before you even get touched, you should be ready to chug that shit like a walking muckbang before anyone can touch you due to the time it takes for the chem to take effect being longer than 0.

Most players do not have the foresight to think theyll be in a situation where they can survive if they had movement speed at the right time and used it at the right time.

This chem is clearly meant to have a very high skill floor that wipes 99% of players from being able to use it.

**Comment by ExcessiveUseOfCobblestone** [comment]
I mean I'd rather not see it used ever than see it made accessible so it's used by every gamer. Mobility is either a flop or a bop, no in between.

**Comment by starrm4nn** [comment]
> I mean I'd rather not see it used ever than see it made accessible so it's used by every gamer. Mobility is either a flop or a bop, no in between.

would changing it so it only drops items that are normal sized or bigger be fine?

**Comment by starrm4nn** [comment]
Couldn't figure out how to make the movement speed modifier scale with purity, It didn't work when i added ``creation_purity`` to ``add_movespeed_modifier`` and ``remove_movespeed_modifier``

File: code/modules/movespeed/modifiers/reagent.dm [hunk 2]
```diff
@@ -5,7 +5,7 @@
 	multiplicative_slowdown = -0.55
 
 /datum/movespeed_modifier/reagent/ephedrine
-	multiplicative_slowdown = -0.35
+	multiplicative_slowdown = -0.5
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/movespeed/modifiers/reagent.dm:8]: If you want to do purity, you'd set `variable = TRUE` here and remove this line, commenting that it gets set on add.

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 3]
```diff
@@ -596,16 +596,14 @@
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm:596]: on L592ish

```DM
/datum/reagent/medicine/ephedrine/on_mob_metabolize(mob/living/affected_mob)
	. = ..()
        var/purity_movespeed_accounting = -0.35*normalise_creation_purity()
        affected_mob.add_or_update_variable_movespeed_modifier(datum/movespeed_modifier/reagent/epinephrine, update = TRUE, purity_movespeed_accounting)

**Review by ExcessiveUseOfCobblestone — COMMENTED** [review]
If you want to add purity see review, otherwise hmu and ill review proper for merge

**Comment by starrm4nn** [comment]
@ExcessiveUseOfCobblestone alright, its done

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 4]
```diff
@@ -589,21 +589,24 @@
...
 	. = ..()
 	affected_mob.remove_movespeed_modifier(/datum/movespeed_modifier/reagent/ephedrine)
 
 /datum/reagent/medicine/ephedrine/on_mob_life(mob/living/carbon/affected_mob, seconds_per_tick, times_fired)
 	. = ..()
-	if(SPT_PROB(10 * (1.5-creation_purity), seconds_per_tick) && iscarbon(affected_mob))
-		var/obj/item/I = affected_mob.get_active_held_item()
-		if(I && affected_mob.dropItemToGround(I))
+	var/obj/item/active_held_item = affected_mob.get_active_held_item()
+	if(SPT_PROB(10 * (1.5-creation_purity), seconds_per_tick) && iscarbon(affected_mob) && active_held_item.w_class > WEIGHT_CLASS_SMALL)
```
Comment (mc-oofert) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm:602]: active held item can be null

Comment (starrm4nn) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm:602]: do i just check if it returns null and set it to 0 or is there a better way of doing this

Comment (ExcessiveUseOfCobblestone) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm:602]: think you can get away with `active_held_item?.w_class`

**Comment by ExcessiveUseOfCobblestone** [comment]
Also feel free to ping me on discord if you are like "I got time on the PR and want to back-forth to get it done"

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm (hunk 4 above)
Comment (starrm4nn) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: thanks

File: code/modules/movespeed/modifiers/reagent.dm [hunk 5]
```diff
@@ -5,7 +5,8 @@
 	multiplicative_slowdown = -0.55
 
 /datum/movespeed_modifier/reagent/ephedrine
-	multiplicative_slowdown = -0.5
+	variable = TRUE
+	// multiplicative_slowdown is set on add
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/movespeed/modifiers/reagent.dm:9]:
```suggestion
	// speed is based on purity of the reagent at the time of metabolization
```

**Pull request merged**